lib/UI/calendar/widgets/logout_button.dart

```diff
@@ -19,28 +20,59 @@ class _LogoutButtonState extends State<LogoutButton> {
     return Padding(
       padding: const EdgeInsets.only(right: 16.0),
       child: InkWell(
-        onTap: () async {
-          if (!_isLoading) {
-            setState(() {
-              _isLoading = true;
-            });
-            await FirebaseAuth.instance.signOut();
-            setState(() {
-              _isLoading = false;
-            });
-            Navigator.pushAndRemoveUntil(
-              context,
-              MaterialPageRoute(builder: (context) => MyApp()),
-              (route) => false,
-            );
-          }
+        onTap: () {
+          _checkShouldSignOut();
         },
         child: _isLoading
-            ? Center(
-                child: CircularProgressIndicator(strokeWidth: 2),
-              )
+            ? Center(child: CircularProgressIndicator(strokeWidth: 2))
             : Icon(Icons.logout),
       ),
     );
   }
+
+  void _navigateToLandingScreen(BuildContext context) {
+    Navigator.pushAndRemoveUntil(
+      context,
+      MaterialPageRoute(builder: (context) => LandingScreen()),
+      (route) => false,
+    );
+  }
+
+  void _checkShouldSignOut() async {
+    if (!_isLoading) {
```

Comment:
Instead of putting the entire function inside of an if, you can do something like this:

```
if (isLoading) return;
// the rest of the method
```